sdk/src/test/java/com/google/cloud/dataflow/sdk/io/TextIOTest.java

```diff
@@ -284,28 +307,71 @@ public void testReadDisplayData() {
       }
     }
 
-    List<String> actual = new ArrayList<>();
+    List<List<String>> actual = new ArrayList<>();
     for (File tmpFile : expectedFiles) {
       try (BufferedReader reader = new BufferedReader(new FileReader(tmpFile))) {
+        List<String> currentFile = Lists.newArrayList();
         for (;;) {
           String line = reader.readLine();
           if (line == null) {
             break;
           }
-          actual.add(line);
+          currentFile.add(line);
         }
+        actual.add(currentFile);
       }
     }
 
-    String[] expected = new String[elems.length];
+    LinkedList<String> expectedElements = Lists.newLinkedList();
```

Comment:
Does this need to explicitly be typed as a `LinkedList`? Did not see that used anywhere.

Should maintain consistent style in constructing lists (L310 vs L313) and types if possible (We use `ArrayList` twice and LinkedList once).